### PR TITLE
Expand cancellation token support to getOccurrences and getType

### PR DESCRIPTION
**Pull request by pittsw**
This change adds cancellation token support to `getOccurrences` and `getType` to improve editor responsiveness when working on large files.

This change also adds some basic perf logging, which I'm fine to either remove or put in a separate PR.

### CODE REVIEW

**Comment by freebroccolo** [comment]
This looks good, thanks! A separate PR for the logging stuff would be nice if you don't mind. Happy to merge both.

**Comment by pittsw** [comment]
Cool, I'll go ahead and send one just for the logging and then once that's merged in I'll update this one :)

File: src/bin/server/session/analyzer.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export default class Analyzer implements server.Disposable {
   ) => Promise<void>) &
     _.Cancelable;
   private readonly bsbDiagnostics: { [key: string]: types.Diagnostic[] } = {};
+  private _refreshCancellationTokenSource: server.CancellationTokenSource = new server.CancellationTokenSource();
```
Comment (arxanas) [inline src/bin/server/session/analyzer.ts]: Why is this called the "refresh" cancellation token?

Comment (pittsw) [reply src/bin/server/session/analyzer.ts]: Oh, because it's used in refreshImmediate and refreshDebounced.  I can rename it.

**Comment by pittsw** [comment]
@freebroccolo: sent a new iteration.

File: src/bin/server/session/analyzer.ts (hunk 1 above)
Comment (arxanas) [reply src/bin/server/session/analyzer.ts]: The name seems fine to me, I just forgot that that was what that was called.

**Pull request merged**